packages/react-query/src/createTRPCReact.tsx

```diff
@@ -174,13 +174,22 @@ export type DecoratedProcedureRecord<
     : never;
 };
 
-export type CreateTRPCReact<TRouter extends AnyRouter, TSSRContext, TFlags> = {
-  useContext(): CreateReactUtilsProxy<TRouter, TSSRContext>;
-  Provider: TRPCProvider<TRouter, TSSRContext>;
-  createClient: CreateClient<TRouter>;
-  useQueries: TRPCUseQueries<TRouter>;
-  useDehydratedState: UseDehydratedState<TRouter>;
-} & DecoratedProcedureRecord<TRouter['_def']['record'], TFlags>;
+type Overwrite<TType, TWith> = Omit<TType, keyof TWith> & TWith;
+
+export type CreateTRPCReact<
+  TRouter extends AnyRouter,
+  TSSRContext,
+  TFlags,
+> = Overwrite<
+  DecoratedProcedureRecord<TRouter['_def']['record'], TFlags>,
+  {
+    useContext(): CreateReactUtilsProxy<TRouter, TSSRContext>;
+    Provider: TRPCProvider<TRouter, TSSRContext>;
+    createClient: CreateClient<TRouter>;
+    useQueries: TRPCUseQueries<TRouter>;
+    useDehydratedState: UseDehydratedState<TRouter>;
+  }
+>;
```

Comment:
It would be even nicer if the intersection gave some sort of error when trying to use it

```ts
type C = Ovewrite<A & B, IntersectionError<A, B, "This key collides with a built-in method, you should rename this router or procedure on your backend">>
```

Reply:
Or maybe it should fail earlier and fail the whole creation of the client, but only if we can showcase _which_ property that is wrong